src/wms/src/main/java/org/geoserver/wms/RenderingVariables.java

```diff
@@ -37,6 +38,12 @@ public static void setupEnvironmentVariables(WMSMapContent mapContent) {
         EnvFunction.setLocalValue(WMS_SRS, mapContent.getRequest().getSRS());
         EnvFunction.setLocalValue(WMS_WIDTH, mapContent.getMapWidth());
         EnvFunction.setLocalValue(WMS_HEIGHT, mapContent.getMapHeight());
+        // add custom env variables
+        Map<String, Object> env = mapContent.getRequest().getEnv();
```

Comment:
Can work, but where are env variables handled now? Doesn't this change make the existing handling reduntant?
Could you elaborate on why this change was necessary to start with?

Reply:
This change was needed to get (geoserver)-side requests with env parameters working at all.
Before this change, server side generated and triggered requests lost their possibly given env parameters.
That may be the reason why the old animator endpoint implementation did not work with env parameters as mentioned by some users on the old blog post about that feature. 

I did not find any other server side code that produces GetMap Requests together with env-variables so this may be a long forgotten bug. Handling of client side requests seems to run through other parts of code which correctly handle env parameters.

I can link the matching parts of code if you need, just give me a sign

Reply:
We use environment injection in multiple places, it's definitely still working. Since it can be used in places other than WMS, it' s handled by a DispatcherCallback:

https://github.com/geoserver/geoserver/blob/6e9e25c0c7cdda9ada9f33f8255130d3afc76801/src/main/src/main/java/org/geoserver/ows/EnviromentInjectionCallback.java#L20

This allows injection across all protocols. However, during the animation bulding, the callback is not invoked, as the frame production does not go through the dispatcher to start with (guess the animator had a similar setup).

This code change is redundant in the WMS code path, could you move it in the animator itself instead?

Reply:
Could you give me a hint on how usage of the dispatcher would look like in this case?

Reply:
ping @aaime : I can see no obvious way how to do this, any suggestions?

Reply:
"This code change is redundant in the WMS code path, could you move it in the animator itself instead?"
I meant, do not change the WMS code path, during normal WMS operations env vars are already working properly, you're addressing a situation that is specific to how the animator ends up calling the WMS classes.

Instead, do those calls inside the download process module, probably inside DownloadMapProcess, somewhere before calling GetMap.run. The issue is specific to gs-wps-download, does not happen in a normal WMS GetMap, so the solution should not mess with the WMS code.